src/software/world/team.cpp

```diff
@@ -129,6 +129,19 @@ void Team::setRobotExpiryBuffer(const Duration& new_robot_expiry_buffer_duration
     robot_expiry_buffer_duration = new_robot_expiry_buffer_duration;
 }
 
+void Team::setUnavailableRobotCapabilities(
+    RobotId id, const std::set<RobotCapability>& new_unavailable_robot_capabilities)
+{
+    for (Robot& robot : team_robots)
```

Comment:
Just a suggestion to avoid repeating code, you could call getRobotById instead of looping through the team again, then set the new capabilities if the getRobotById returns a robot, otherwise do nothing.

Reply:
I actually used this approach at the beginning but then realized that `getRobotById` returns a copy of the robot with that id, so updating it would not update the actual robot. That's why I decided to add `setUnavailableRobotCapabilities` so it can update the actual copy of the robot.

Reply:
Oh that makes sense, thanks for clarifying!